geode-web-management/src/distributedTest/java/org/apache/geode/management/client/GetStartingMemberTest.java

```diff
@@ -0,0 +1,89 @@
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.client;
+
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.context.WebApplicationContext;
+
+import org.apache.geode.management.api.ClusterManagementException;
+import org.apache.geode.management.api.ClusterManagementGetResult;
+import org.apache.geode.management.api.ClusterManagementService;
+import org.apache.geode.management.api.RestTemplateClusterManagementServiceTransport;
+import org.apache.geode.management.configuration.Member;
+import org.apache.geode.management.internal.rest.LocatorWebContext;
+import org.apache.geode.management.internal.rest.PlainLocatorContextLoader;
+import org.apache.geode.management.runtime.MemberInformation;
+import org.apache.geode.test.dunit.rules.ClusterStartupRule;
+import org.apache.geode.test.junit.rules.ConcurrencyRule;
+
+@RunWith(SpringRunner.class)
+@ContextConfiguration(locations = {"classpath*:WEB-INF/management-servlet.xml"},
+    loader = PlainLocatorContextLoader.class)
+@WebAppConfiguration
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
+public class GetStartingMemberTest {
+
+  @Autowired
+  private WebApplicationContext webApplicationContext;
+
+  @Rule
+  public ClusterStartupRule cluster = new ClusterStartupRule(1);
+
+  @Rule
+  public ConcurrencyRule concurrency = new ConcurrencyRule();
+
+  private ClusterManagementService client;
+  private LocatorWebContext webContext;
+
+  @Before
+  public void before() {
+    cluster.setSkipLocalDistributedSystemCleanup(true);
+    webContext = new LocatorWebContext(webApplicationContext);
+    client = new ClusterManagementServiceBuilder().setTransport(
+        new RestTemplateClusterManagementServiceTransport(
+            new RestTemplate(webContext.getRequestFactory())))
+        .build();
+  }
+
+  @Test
+  public void getStartingMember() throws Exception {
+    Member server0 = new Member();
+    server0.setId("server-0");
+    concurrency.add(() -> cluster.startServerVM(0, webContext.getLocator().getPort()));
+    concurrency.add(() -> {
+      ClusterManagementGetResult<Member, MemberInformation> result = null;
+      while (result == null) {
+        try {
+          result = client.get(server0);
+        } catch (ClusterManagementException e) {
+          // this is expected for the first several tries
+        }
+      }
+      return result;
+    });
+    concurrency.executeInParallel();
```

Comment:
ExecutorServiceRule would really clean this up and I think we should look into merging these two rules. In general, I think ExecutorServiceRule is preferable (I can explain outside of PR). If you switch to ExecutorServiceRule the test could use an assertion in await and seems much more readable to me:
```
import static java.util.concurrent.TimeUnit.MILLISECONDS;
import static org.apache.geode.test.awaitility.GeodeAwaitility.await;
import static org.assertj.core.api.Assertions.assertThat;

  @Rule
  public ExecutorServiceRule executorServiceRule = new ExecutorServiceRule();

  @Test
  public void getStartingMember() throws Exception {
    Member server0 = new Member();
    server0.setId("server-0");

    Future<Void> startServer = executorServiceRule.submit(() -> {
      cluster.startServerVM(0, webContext.getLocator().getPort());
    });

    await()
        .ignoreException(ClusterManagementException.class)
        .untilAsserted(() -> assertThat(client.get(server0)).isNotNull());

    startServer.get(getTimeout().toMillis(), MILLISECONDS);
  }
```

Reply:
Great, thank you so much!